<launch>
    <!-- Launch Arguments -->
    <arg name="joy_dev_input" default="/dev/input/js0" />
    <arg name="use_deadman" default="false" />
    <arg name="use_turbo" default="false" />
    <arg name="with_telemetry_gui" default="false" />

    <!-- Husky Gazebo Launch File -->
    <include file="$(find husky_gazebo)/launch/husky_playpen.launch"/>

    <!-- joy Node -->
    <node respawn="true" pkg="joy" type="joy_node" name="husky_joy">
        <param name="dev" type="string" value="$(arg joy_dev_input)"/>
        <param name="deadzone" value="0.12"/>
    </node>

    <!-- Axes -->
    <param name="axis_linear" value="1" type="int"/>
    <param name="axis_angular" value="2" type="int"/>
    <param name="scale_linear" value="1" type="double"/>
    <param name="scale_angular" value="0.75" type="double"/>
    <param name="scale_turbo" value="1.5" type="double"/>

    <!-- Teleop Node -->
    <node pkg="husky_teleop" type="husky_teleop_turbo" name="teleop_turbo">
        <param name="turbo_allowed" type="bool" value="$(arg use_turbo)" />
        <param name="deadman_required" type="bool" value="$(arg use_deadman)" />
    </node>

    <!-- Teleop Telemetry GUI Node -->
    <node pkg="husky_teleop"
          type="husky_teleop_telemetry_gui"
          name="teleop_telemetry_gui"
          if="$(arg with_telemetry_gui)" />

</launch>
